### DISCUSSION TITLE
Some feature requests

### DISCUSSION THREAD

**Discussion by johngarciava** [trakt/showly#661] [Ideas / Feature Requests]
**TL;DR** Reasons why I still use Trakt app 10% (90% Showly) of the time: anti-spoiler/blur settings, movies and additional play tracking, favoriting services.

1. Add addtional plays
     - If you already watched a movie and you click on "added to my movies" it basically removes it on Showly as watched and if you have Trakt synced it'll ask if you want it removed on there too. I feel like it should work the way Trakt does where if you click on the check, it gives multiple options: Watching now, Just finished, Release date, Other date. I think this button on Showly should just say "Add a play" or "track this movie" instead of "Add to my movies" since that can be confusing, and have another button right next to it where it goes to the history of plays (Trakt app has this too). If the movie has a play, the button that should say "X plays" (right now it says added to my movies, kind of confusing if you're supporting multiple plays) This whole thing is definitely something I think the Trakt mobile app handles way better (tbf Showly literally can't lol). For now, I think when you track a movie on Showly, giving the option of Just Finished and Other date would be a massive improvement. Handling movies on the Trakt app is definitely a better experience than Showly, but additional plays with shows are also a thing that a lot of people do, even though personally I rewatch movies most of the time.

2. Anti-spoiler settings
     - Like hide/blur ratings that are the same or similar to Trakt's implementation. Basically blurs ratings of unwatched shows/movies app-wide, can just tap the rating to unblur if you really want to see it. I honestly find it odd Trakt is the only one who offers this feature on both their website and mobile app, and of the 100s of other tracking apps, none of them implement it, it's so underrated and useful. 

3. Being able to "favorite" streaming services.
     - Currently, if you check on a movie/shows details, it'll show all the services that movie is on, so you have to scroll a lot. However, if something is available on Amazon Video, I don't really care because I'm never going to rent/buy something through that. There might be cases where I want to see all the services, say it's a movie I've been dying to watch, but for most uses, I'd like to favorite streaming services so that there can be an option where only services that I have get displayed instead. Both Trakt's website and their app has this as well, their UI allows for this switching between favorite and all services quickly, but the way Showly displays it definitely makes it trickier. Can also use this for Showly's Discover tab for example, where you can have it only show stuff from services you have favorited if you like. 

4. Notifications for show/movie availability on a favorited service
     - Now a feature that Trakt doesn't have for sure is a feature where the app notifies you when a movie becomes available on a streaming app you have favorited. Say I want to watch The Matrix trilogy and the first 2 are on Netflix, but the 3rd is only available for rent. However with the feature I'll be able to add the 3rd movie to a watchlist, or maybe just enabled with a button, anything that basically says, notify me when this movie is available to watch on any streaming service I have. The Ava - Movie Assistant app on Android has this feature and I would love to see it implemented on Showly too!

5. Info on related shows/movies of franchises on its details page
     - Another feature request that I think would be really nice (if possible) is if you go to a movie's page, and that movie is part of a series of movies, add a section "this movie is a part of a collection (duology, trilogy, several movies)" or something similar in the movie's page, where if the user clicks on it, it'll show all the related movies in order of release date. Or, another way of implementing it, instead of the user having to click and it taking the user to another page, just like how on the movie page there's a "You may also like:" section, add another section that says "This movie is part of a collection:" Showing the related movies right below it, but in this one you add the year of release date below the movie posters and order it ascendingly. I really like idea of this feature because this is something I've done SO MANY times when I want to know how many movies and shows does a franchise have, where I look up on google a certain movie of a franchise, say the first harry potter movie, and then on its wikipedia, usually at the top it mentions the movie's prequels and sequels and all that related stuff if it has any. Would be great to be able to do this on Showly with its much superior UI and UX. It can also maybe cover shows, like game of thrones with the house of the dragon prequel, or movies that have related shows (LOTR). 

6. Possible iOS version?
   - Not really a feature suggestion but I think I've seen you say you would never do an iOS version? If that's the case is it OK if I go on Reddit or something and beg developers to do an iOS version of Showly and if somebody does come up to the plate would you be willing to work with them?

I've recently moved to an iPhone, and Showly is honestly literally the only thing I miss about Android lol. I still always have my old OnePlus 5 on hand strictly because of this app. Showly's Progess tab page specifically is just unparalleled when it comes to tracking your shows and no iOS (nor android) app comes anywhere near close to this level of excellent UI & UX execution! The Upcoming and On Hold categories are chef's kiss.  Would really be incredible to see this app on iOS.

**Comment by Tomekske** [comment]
Maybe the community can collaborate on a Flutter app? One codebase for all platforms. I'd gladly work on a potential Flutter rewrite. It may attract also other developers since Flutter is booming right now, so potential bugs, issues and features might be solved faster
